ocs_ci/ocs/platform_nodes.py

```diff
@@ -432,3 +512,390 @@ def restart_nodes_teardown(self):
         # Start the instances
         if stopped_instances:
             self.aws.start_ec2_instances(instances=stopped_instances, wait=True)
+
+    def attach_nodes_to_cluster(self, node_list):
+        """
+        Attach nodes in the list to the cluster
+
+        Args:
+            node_list (list): of AWSUPINode/AWSIPINode objects
+
+        """
+        if self.deployment_type.lower() == 'upi':
+            self.attach_nodes_to_upi_cluster(node_list)
+
+    def attach_nodes_to_upi_cluster(self, node_list):
+        """
+        Attach node to upi cluster
+
+        Args:
+            node_list (list): of AWSUPINode objects
+
+        """
+        if node_list[0].node_type == 'RHEL':
+            self.attach_rhel_nodes_to_upi_cluster(node_list)
+
+    def attach_rhel_nodes_to_upi_cluster(self, node_list):
+        """
+        Attach RHEL nodes to upi cluster
+
+        Args:
+            node_list (list): of AWSUPINode objects with RHEL os
+
+        """
+        rhel_pod_name = "rhel-ansible"
+        rhel_pod_obj = create_rhelpod(
+            constants.DEFAULT_NAMESPACE, rhel_pod_name, 600
+        )
+        timeout = 4000  # For ansible-playbook
+
+        # copy openshift-dev.pem to RHEL ansible pod
+        pem_src_path = "~/.ssh/openshift-dev.pem"
+        pem_dst_path = "/openshift-dev.pem"
+        pod.upload(rhel_pod_obj.name, pem_src_path, pem_dst_path)
+        repo_dst_path = constants.YUM_REPOS_PATH
+        repo = os.path.join(
+            constants.REPO_DIR, f"ocp_{get_ocp_version('_')}.repo"
+        )
+        assert os.path.exists(repo), f"Required repo file {repo} doesn't exist!"
+        repo_file = os.path.basename(repo)
+        pod.upload(
+            rhel_pod_obj.name, repo, repo_dst_path
+        )
+        # copy the .pem file for our internal repo on all nodes
+        # including ansible pod
+        # get it from URL
+        mirror_pem_file_path = os.path.join(
+            constants.DATA_DIR,
+            constants.INTERNAL_MIRROR_PEM_FILE
+        )
+        dst = constants.PEM_PATH
+        pod.upload(rhel_pod_obj.name, mirror_pem_file_path, dst)
+        # Install scp on pod
+        rhel_pod_obj.install_packages("openssh-clients")
+        # distribute repo file to all RHEL workers
+        hosts = [
+            node.aws_instance_obj.private_dns_name for node in
+            node_list
+        ]
+        # Check whether every host is acceptin ssh connections
+        for host in hosts:
+            self.check_connection(rhel_pod_obj, host, pem_dst_path)
+
+        for host in hosts:
+            disable = "sudo yum-config-manager --disable *"
+            rhel_pod_obj.exec_cmd_on_node(
+                host, pem_dst_path, disable, user=constants.EC2_USER
```

Comment:
@shylesh can you check once over VMware I guess constant.EC2_USER will not work User is constants.VM_RHEL_USER for rhel VM in VMware

Reply:
@prsurve , This will be for AWSUPI RHEL nodes, VMware function is not implemented yet.